homeassistant/components/homee/config_flow.py

```diff
@@ -0,0 +1,243 @@
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: HomeeConfigEntry,
+    ) -> SchemaOptionsFlowHandler:
+        """Get the options flow handler."""
+        return SchemaOptionsFlowHandler(config_entry, OPTIONS_FLOW)
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        # self.homee_host: str = None
+        # self.homee_id: str = None
+        self.homee: Homee = None
+        self.all_devices: bool = True
+        self.debug_data: bool = False
+
+    async def async_step_user(self, user_input=None) -> ConfigFlowResult:
+        """Handle the initial user step."""
+
+        errors = {}
+        if user_input is not None:
+            try:
+                self.homee = await validate_and_connect(self.hass, user_input)
+                await self.async_set_unique_id(self.homee.settings.uid)
+                self._abort_if_unique_id_configured()
+                _LOGGER.info(
+                    "Created new homee entry with ID %s", self.homee.settings.uid
+                )
+                self.debug_data = user_input[CONF_ADD_HOMEE_DATA]
+                return await self.async_step_groups()
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except AbortFlow:
+                return self.async_abort(reason="already_configured")
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=AUTH_SCHEMA,
+            errors=errors,
+        )
+
+    async def async_step_groups(self, user_input=None) -> ConfigFlowResult:
+        """Configure groups options."""
+        groups_selection = {
+            str(g.id): f"{g.name} ({len(g.nodes)})" for g in self.homee.groups
+        }
+
+        # There doesn't seem to be a way to disable a field - so we need 2 separate versions.
+        GROUPS_SCHEMA = vol.Schema(
+            {
+                vol.Required(
+                    CONF_WINDOW_GROUPS,
+                    default=[],
+                ): cv.multi_select(groups_selection),
+                vol.Required(
+                    CONF_DOOR_GROUPS,
+                    default=[],
+                ): cv.multi_select(groups_selection),
+            }
+        )
+
+        if user_input is not None:
+            return self.async_create_entry(
+                title=f"{self.homee.settings.uid} ({self.homee.host})",
+                data={
+                    CONF_HOST: self.homee.host,
+                    CONF_USERNAME: self.homee.user,
+                    CONF_PASSWORD: self.homee.password,
+                },
+                options={
+                    CONF_ADD_HOMEE_DATA: self.debug_data,
```

Comment:
What are groups?

Reply:
Groups are groups of devices, that are configurable in Homee.
Here they are used to work around a deficiency in the Homee Data model, that reports us Door & Window sensors only as Binary sensors. So that the user can have the correct types of sensors set in HA, we need to manually put them in corresponding groups in Homee and then tell HA, which Group contains Window- and which Door Sensors.
I long wanted to get rid of this, but didn't find a way.

Reply:
But door and window sensors are just binary sensors right?

Reply:
Yeah, but they get different device classes here: https://github.com/Taraman17/hass-homee/blob/7285830f7f0ece68bd941137b3652b76709815dd/custom_components/homee/binary_sensor.py#L184

Reply:
But this is for binary sensor right? Since this PR is only implementing cover, let's postpone this discussion and remove it for now

Reply:
OK.